pkg/auth/authz_utils.go

```diff
@@ -0,0 +1,32 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/pkg/ocm"
+	"github.com/pkg/errors"
+)
+
+func ValidateAccessToCPUArchitecture(ctx context.Context, authzHandler Authorizer, cpuArchitecture string) error {
+	var err error
+	var armArchAllowed bool
+
+	switch cpuArchitecture {
+	case common.ARM64CPUArchitecture:
+
+		armArchAllowed, err = authzHandler.HasOrgBasedCapability(ctx, ocm.ArmCapabilityName)
```

Comment:
not crucial for now, but lines 20-27 can be extracted to a generic internal func like hasArchSupport(capabilityName, arch string). we can also handle it later when adding another arch capability.

Reply:
yeah I've just meant to extract these to a generic func, so adding another arch to the case would be simpler. can be then later though.